lib/Dancer2/Core/Error.pm

```diff
@@ -255,8 +254,13 @@ has content => (
                     }
                 );
             };
+
             # return rendered content unless there was an error.
-            defined $content && return $content;
+            if (defined $content) {
+                # Why is this necessary here and not with other templates e.g. like route templates?
+                utf8::encode($content) if utf8::is_utf8($content); # less confusing via String::UnicodeUTF8: $content = get_utf8($content) if is_unicode($content)
```

Comment:
It was needed in an app that was set to UTF8. We could only do it if the config is set to UTF8 to make it that much safer I suppose. I'll add that ASAP.

I also agree that it probably points to another bug that needs fixed that was never seen since the error feature has been broken. If that gets fixed this part will still be good to go.

Sent from my iPhone

> On Feb 14, 2015, at 7:54 AM, Russell Jenkins <notifications@github.com> wrote:
> 
> In lib/Dancer2/Core/Error.pm:
> 
> >              # return rendered content unless there was an error.
> > -            defined $content && return $content;
> > +            if (defined $content) {
> > +                # Why is this necessary here and not with other templates e.g. like route templates?
> > +                utf8::encode($content) if utf8::is_utf8($content); # less confusing via String::UnicodeUTF8: $content = get_utf8($content) if is_unicode($content)
> You can configure an app to use an encoding other than UTF-8. The "Why is it necessary" comment and having to do that twice points to this being the wrong place to be dealing with encodings.
> 
> For the content returned from a route (including templates), that content is added to the response object then the response is asked to encode it via $response->content($content); $response->encode_content;.
> 
> Dancer2::Core::Error->throw populates the response. Do any encoding there.
> 
> —
> Reply to this email directly or view it on GitHub.
>